### PR TITLE
DHT fs read commands

### PR DESCRIPTION
**Pull request by toomyem**
Added two debug commands for DHT sensor:
/drivers/dht/read <pin name>
/drivers/dht/read_11 <pin name>

### CODE REVIEW

File: src/config_default.h [hunk 1]
```diff
@@ -1122,6 +1122,17 @@
 #    endif
 #endif
 
+/**
+ * Debug file system command to read DHT sensor on the bus.
+ */
+#ifndef CONFIG_DHT_COMMAND_READ
+#    if defined(BOARD_ARDUINO_NANO) || defined(BOARD_ARDUINO_UNO) || defined(BOARD_ARDUINO_PRO_MICRO) || defined(CONFIG_MINIMAL_SYSTEM)
```
Comment (toomyem) [inline src/config_default.h]: I was not sure about those conditions - just copied them after _ds18b20_ above.

File: src/drivers/sensors/dht.c [hunk 2]
```diff
@@ -164,24 +174,6 @@ static int read_isr(struct dht_driver_t *self_p, uint8_t *buf_p)
     return (res);
 }
 
-int dht_module_init()
```
Comment (toomyem) [inline src/drivers/sensors/dht.c]: Moved this function down, below 'static' section.

File: src/config_default.h (hunk 1 above)
Comment (eerimoq) [reply src/config_default.h]: These Arduino boards have very limited amount of memory, and lots of fs commands are disabled by default.

File: src/drivers/sensors/dht.c [hunk 3]
```diff
@@ -210,7 +202,111 @@ static int read_sensor(struct dht_driver_t *self_p, uint8_t *buf_p)
...
+                       const char *argv[],
+                       void *chout_p,
+                       void *chin_p,
+                       void *arg_p,
+                       void *call_arg_p)
+{
+    return cmd_read_sensor(argc, argv, chout_p, dht_read, "read");
+}
+
+static int cmd_11_read_cb(int argc,
```
Comment (eerimoq) [inline src/drivers/sensors/dht.c]: Rename to `cmd_read_11_cb`.

File: src/drivers/sensors/dht.c [hunk 4]
```diff
@@ -47,6 +47,16 @@
 #define TEMP_LSB_INDEX                              3
 #define PARITY_INDEX                                4
 
+struct module_t {
+    int8_t initialized;
+#if CONFIG_DHT_COMMAND_READ == 1
+    struct fs_command_t cmd_read;
+    struct fs_command_t cmd_11_read;
```
Comment (eerimoq) [inline src/drivers/sensors/dht.c]: Rename to `cmd_read_11`.

File: src/drivers/sensors/dht.c [hunk 5]
```diff
@@ -210,7 +202,111 @@ static int read_sensor(struct dht_driver_t *self_p, uint8_t *buf_p)
...
+    if (argc != 2) {
+        std_printf(OSTR("Usage: %s <pin name>\r\n"), fn_name_p);
+
+        return (-EINVAL);
+    }
+
+    pin = board_pin_string_to_device_index(argv[1]);
+
+    if (pin < 0) {
+        std_printf(OSTR("Invalid pin name.\r\n"));
```
Comment (eerimoq) [inline src/drivers/sensors/dht.c]: Add the pin name (`args[1]`) to the string to help the user understand which argument was wrong.

File: src/drivers/sensors/dht.c [hunk 6]
```diff
@@ -210,7 +202,111 @@ static int read_sensor(struct dht_driver_t *self_p, uint8_t *buf_p)
...
+
+        return (-ENODEV);
+    }
+
+    dht_init(&dht, &pin_device[pin]);
+    
+    res = read_p(&dht, &temperature, &humidity);
+
+    if (res == 0) {
+        std_fprintf(chout_p, "Temperature: %f, Humidity: %f\r\n", temperature, humidity);
```
Comment (eerimoq) [inline src/drivers/sensors/dht.c]: Add `OSTR()` for output format strings. It makes it possible to replace format strings when pre-processing source files. SOAM does this, and it saves a lot of memory for applications with lots of output strings.

File: src/drivers/sensors/dht.c [hunk 7]
```diff
@@ -210,7 +202,111 @@ static int read_sensor(struct dht_driver_t *self_p, uint8_t *buf_p)
...
+}
+
+static int cmd_read_cb(int argc,
+                       const char *argv[],
+                       void *chout_p,
+                       void *chin_p,
+                       void *arg_p,
+                       void *call_arg_p)
+{
+    return cmd_read_sensor(argc, argv, chout_p, dht_read, "read");
```
Comment (eerimoq) [inline src/drivers/sensors/dht.c]: Add `(...)` around the returned value. It's the code style.

**Comment by eerimoq** [comment]
Please add tests of the commands. There are fs commands tested in other modules, so there are plenty of examples.

**Pull request merged**